packages/cloud_firestore/ios/Classes/CloudFirestorePlugin.m

```diff
@@ -477,8 +549,26 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
         result(getFlutterError(error));
         return;
       }
-      result(parseQuerySnapshot(snapshot));
+      [self.channel invokeMethod:@"DocumentSnapshot"
+                       arguments:@{
+                         @"handle" : handle,
+                         @"path" : snapshot ? snapshot.reference.path : [NSNull null],
+                         @"data" : snapshot && snapshot.exists ? snapshot.data : [NSNull null],
```

Comment:
nit: `snapshot && snapshot.exists ?` can be replaced as `snapshot.exists ?` since if snapshot is nil, snapshot.exists will also return false.
```suggestion
                         @"data" : snapshot.exists ? snapshot.data : [NSNull null],
```